### ISSUE TITLE
pass custom `--baseurl` argument to jekyll serve?

### ISSUE THREAD

**Issue by cboettig** [yihui/servr#14]
tl;dr  The `jeykll` function cannot actually an argument for `baseurl` and pass it to `servr_config` because you have an argument of the same name in `dynamic_site`.  Perhaps the latter argument could be renamed to avoid this conflict?  (Or maybe you have a more clever solution).  

The [official Jekyll Documentation recommends](http://jekyllrb.com/docs/github-pages/#project-page-url-structure) using an empty string to `--baseurl` when previewing a site that is deployed on a gh-pages project repo (as opposed to a user/organization repo like `username.github.io`).  With the appropriate configuration this makes it easy to preview the site locally and still deploy it automatically on gh-pages.

Looks like I should just be able to set `baseurl=""`, but this results in an error: `formal argument "baseurl" matched by multiple actual arguments` since baseurl is also an argument to `dynamic_site()`.  If I understand correctly, I only want to change the baseurl being used by `server_config()` and not that one being used by `dynamic_site`.   Seems like renaming the argument as it is used in the latter case would do the trick?

**Comment by yihui** [comment]
I do not quite see the point of providing an empty `--baseurl`. It makes the URL `http://localhost:4000` instead of `http://localhost:4000/project/`, but I do not see what harm the latter does. You will eventually see `http://foo.github.io/project/` after all. I have not got any problems without setting `baseurl = ""` in `jekyll()`:

- https://github.com/yihui/knitr/blob/gh-pages/_config.yml
- https://github.com/yihui/knitr-jekyll/blob/gh-pages/_config.yml

So I'd say the point on `--baseurl` in the official documentation is probably a red herring...

**Comment by cboettig** [comment]
Okay, thanks for setting me straight, everything seems to work just fine as you say and I agree it looks like the documentation is just a red herring at this point.  (For some reason I recall yesterday having to remove the baseurl string from the URL to get my posts to resolve, but cannot replicate that now).  

Still seems like either the all `servr_config()` options should be available to `jekyll()` or that the documentation should say that `baseurl` is not, but that's pretty minor.  Thanks again
